ArdusatSDK.cpp

```diff
@@ -999,74 +1105,28 @@ const char * Pressure::toJSON(const char * sensorName) {
  * @brief   Constructs RGBLight sensor object, default uses TCS34725 sensor
  * @ingroup rgblight
  *
- * @param   sensor_id Specifies which RGB sensor to use: TCS34725 or ISL29125
- *     - SENSORID_TCS34725 (Default)
- *     - SENSORID_ISL29125
- *
  * Example Usage:
  * @code
  *     RGBLight rgb;                          // Instantiate sensor object
  *     rgb.begin();                           // Initialize sensor
  *     Serial.println(rgb.readToJSON("rgb")); // Read and print values in JSON
  * @endcode
  *****************************************************************************/
-RGBLight::RGBLight(sensor_id_t sensor_id) :
-  islIntensity(CFG1_10KLUX),
-  tcsIt(TCS34725_INTEGRATIONTIME_154MS),
-  tcsGain(TCS34725_GAIN_1X)
+RGBLight::RGBLight(void) :
+  RGBLight(SENSORID_TCS34725, CFG1_10KLUX, TCS34725_INTEGRATIONTIME_154MS, TCS34725_GAIN_1X)
 {
-  this->initializeHeader(sensor_id, DATA_UNIT_LUX, rgblight_sensor_name);
 }
 
 /**
- * @brief   Sets the intensity configuration variable
+ * @brief   Protected constructor used by public constructors
  * @ingroup rgblight
- *
- * @param islIntensity Advanced configuration for ISL29125 intensity
- *     - CFG1_375LUX
- *     - CFG1_10KLUX (Default)
  */
-void RGBLight::configure(uint8_t islIntensity) {
-  if (this->header.sensor_id == SENSORID_ISL29125) {
-    this->islIntensity = islIntensity;
-    this->initialized = false;
-  }
-}
-
-/**
- * @brief   Sets the integration time configuration variable
- * @ingroup rgblight
- *
- * @param tcsIt Advanced configuration for TCS34725 integration time
- *     - TCS34725_INTEGRATIONTIME_2_4MS
- *     - TCS34725_INTEGRATIONTIME_24MS
- *     - TCS34725_INTEGRATIONTIME_50MS
- *     - TCS34725_INTEGRATIONTIME_101MS
- *     - TCS34725_INTEGRATIONTIME_154MS (Default)
- *     - TCS34725_INTEGRATIONTIME_700MS
- */
-void RGBLight::configure(tcs34725IntegrationTime_t tcsIt) {
-  if (this->header.sensor_id == SENSORID_TCS34725) {
-    this->tcsIt = tcsIt;
-    this->initialized = false;
-  }
-}
-
-/**
- * @brief   Sets the gain configuration variable
- * @ingroup rgblight
- *
- * @param tcsGain Advanced configuration for TCS34725 gain
- *     - TCS34725_GAIN_1X (Default)
- *     - TCS34725_GAIN_4X
- *     - TCS34725_GAIN_16X
- *     - TCS34725_GAIN_60X
- */
-void RGBLight::configure(tcs34725Gain_t tcsGain) {
-  if (this->header.sensor_id == SENSORID_TCS34725) {
-    this->tcsGain = tcsGain;
-    this->initialized = false;
-  }
+RGBLight::RGBLight(sensor_id_t sensor_id, uint8_t islIntensity, tcs34725IntegrationTime_t tcsIt, tcs34725Gain_t tcsGain) :
+  islIntensity(islIntensity),
```

Comment:
I was envisioning the generic `RGBLight` class to only have _one_ of the light drivers in it (the default one, in this case, the TCS34725), and no support for non-default configure params in the constructor. Then the class `RGBLightTCS` is almost a no-op class, except for adding an optional constructor that takes config arguments. But the `RGBLightISL` overwrites `readSensor`, and `begin`, etc. with ISL versions. That way you don't need `if` statements in the methods defined here, and you don't have to check that your `sensor_id` is a particular value.